gloo/allreduce_bcube.h

```diff
@@ -16,6 +16,10 @@
 #include <iostream>
 #include <unordered_map>
 
+#ifdef _WIN32
```

Comment:
This doesn't need to be conditionally guarded, it's required for correctness. Other platforms just ended up including it as part of other STL headers.